### ISSUE TITLE
linux-tkg: option to use hardened/ccs patches

### ISSUE THREAD

**Issue by soredake** [Tk-Glitch/PKGBUILDS#222]
And enable apparmor.
Relevant links:
https://aur.archlinux.org/packages/linux-hardened-ccs-apparmor/
https://www.archlinux.org/packages/extra/x86_64/linux-hardened/
https://github.com/anthraxx/linux-hardened
https://osdn.net/projects/tomoyo/

**Comment by Joshua-Ashton** [comment]
AppArmor/""Hardened Patches"" just make people want to tear their hair out constantly and don't really make anything more secure, despite their names.

If you've ever worked with Android before, you will know the absolute suffering this shit causes when you want to do *anything*.

**Comment by Faalagorn** [comment]
While I don't use nor the AppArmor nor the Hardened patches myself, there are a few knobs I've take a note of when messing with the kernel options in menuconfig – I don't use all of them, but some may be a good idea  to leave behind a switch eventually – or even go the more extreme way and disable some of the mitigations for even more performance behind a switch (though with 5.2(?) and forward it could possibly be easier to just add `mitigations=off` to kernel parameters (and on lower ones you can achieve the same, though with a couple of parameters). I also mentioned that briefly on Discord with TkG, but didn't really wanted to push it myself :).

Either way, here's the dump of potentially "security"-affecting parameters I saved for myself on Google Keep, sometimes with a rough note of mine, often paraphrasing what Kconfig help says:
```
CONFIG_IKCONFIG=n – y on arch
CONFIG_BPF_JIT_ALWAYS_ON=y (?)
CONFIG_SLAB_MERGE_DEFAULT=n
CONFIG_SLAB_FREELIST_RANDOM=y
CONFIG_SLAB_FREELIST_HARDENED=y (minor performance sacrifices)
CONFIG_RETPOLINE=y (kernel may run slower)
CONFIG_KEXEC=n
CONFIG_RANDOMIZE_BASE=y https://www.phoronix.com/scan.php?page=news_item&px=KASLR-Default-Linux-4.12
CONFIG_INTEL_TXT=y (?)
CONFIG_FORTIFY_SOURCE=y
CONFIG_SECURITY_ (yama etc.)
CONFIG_SECURITY_SAFESETID=y
```

There's also maybe not so up to date but still interesting overview of some options at https://kernsec.org/wiki/index.php/Kernel_Self_Protection_Project/Recommended_Settings

**Comment by Tk-Glitch** [comment]
I'll think about the best way to handle that. I had other related stuff cooking that will need to be re-thought. I will be posting updates and/or questions in here if required.

**Comment by ghost-101** [comment]
``CONFIG_PAGE_TABLE_ISOLATION`` (KPTI) is also a mitigation that can be disabled for an uplift. Personally, I run my kernels with no mitigations, I instead utilize firejail to its full extent with a bunch of custom profiles, lowering attack surface in Firefox (uMatrix), and not giving root access whenever possible (which firejail also helps block off). This new mitigations=off option is going to be very nice, I didn't know about CONFIG_RANDOMIZE_BASE.

As a side note, I updated Arch's GCC to 9.1.0, and the kernel started complaining about the compiler not having retpoline support, anyone else?

**Comment by Tk-Glitch** [comment]
I think we can close this as we have the Arch hardened linux config and patchset in now so the initial request is fulfilled. Further discussions or problem encountered with these changes are worth a different issue.

**Comment by Ryujinra** [comment]
I very much appreciated the inclusion of these patches as this was the only performance-oriented custom kernels to offer them; however, I see they haven't been included since 5.2. Is there just a delay before certain patches are added to the newer kernels, or have they unfortunately been dropped for good?

**Comment by Tk-Glitch** [comment]
There's indeed a delay, and it'll be added as it gets rebased
